pkgs/build-support/fetchgit/nix-prefetch-git

```diff
@@ -300,7 +300,9 @@ clone_user_rev() {
     local rev="${3:-HEAD}"
 
     if [ -n "$fetchLFS" ]; then
-        HOME=$TMPDIR
+        tmpHomePath="$(mktemp -d "${TMPDIR:-/tmp}/nix-prefetch-git-tmp-home-XXXXXXXX")"
```

Comment:
```suggestion
        tmpHomePath="$(mktemp -d "$TMPDIR/nix-prefetch-git-tmp-home-XXXXXXXX")"
```
TMPDIR is always populated.

Reply:
Hi @SuperSandro2000,

>TMPDIR is always populated.

I tested in NixOS and Ubuntu. It's not set there.

Reply:
You are right but that reveals that yo use mktemp wrong here.

```suggestion
        tmpHomePath="$(mktemp -d nix-prefetch-git-tmp-home-XXXXXXXX)"
```
or this
```suggestion
        tmpHomePath="$(mktemp -d)"
```
is the correct syntax

Reply:
I see. This code snippet is also based on how things already are in the script. I suggest we make another PR to update the use of mktemp/TMPDIR for all occurrences.

Reply:
Sure, go ahead but please don't introduce a new wrong one.